[Pending] Add custom magnetometer orienation documentation (#2135)
PX4/PX4-user_guide: en/assembly/mount_gps_compass.md

## Compass Orientation

> The compass can also be mounted in any _supported_ orientation, where the supported orientations can be seen in [CAL_MAGn_ROT](../advanced_config/parameter_reference.md#CAL_MAG1_ROT), and uses the same convention for [orienting the flight controller](../config/flight_controller_orientation.md#orientation-definition).

Reviewer: "other" was deliberate - we want them to know that if they mount as recommended in the very first para of the doc they can ignore the rest. 

This text is OK too, but what that makes me think is that perhaps we should "spell this out" in the first paragraph - i'e if you do this don't bother reading the rest.

Reviewer: I'll just do it :-)

If you're using the normal [Compass Calibration](../config/compass.md) process the parameter [CAL_MAG_ROT_AUTO](../advanced_config/parameter_reference.md#CAL_MAG_ROT_AUTO) is enabled, and the orientation should be detected automatically.

> If your compass can't be oriented in any of the supported orientations, you can utilize `Custom rotation`, which is documented below.

Reviewer: ```suggestion
If your compass can't be oriented in any of the supported orientations, you can use a [custom rotation](@custom_magnetometer_rotation) as documented below.
```

:::warning

> If you mount the compass at an orientation that isn't supported, for example `Yaw 30`, PX4 will detect the closest supported value. This will result in errors/warnings, even if the calibration appeared to succeed.

Reviewer: ```suggestion
If you mount the compass at an orientation that isn't supported, for example `Yaw 30`, PX4 will detect the closest supported value.
This may result in errors/warnings, even if the calibration appeared to succeed.
```

Author: Yes! I also thought this part isn't so necessary :+1:

:::

### Custom magnetometer orientation

> By selecting the `Custom Euler Angle` in the orientation setting for your magnetometer, you can specify any rotation using [CAL_MAGn_ROLL](../advanced_config/parameter_reference.md#CAL_MAG1_ROLL), [CAL_MAGn_PITCH](../advanced_config/parameter_reference.md#CAL_MAG1_PITCH) and [CAL_MAGn_YAW](../advanced_config/parameter_reference.md#CAL_MAG1_YAW) parameters.
> 
> The roll/pitch/yaw are defined as documented [here](../config/flight_controller_orientation.md#orientation-definition).
> 
> To configure the custom orientation correctly, do the following:
> 1. Set the [CAL_MAGn_ROT](../advanced_config/parameter_reference.md#CAL_MAG1_ROT) parameter to `Custom Euler Angle`
> 2. Set the magnetometer's custom roll/pitch/yaw appropriately
> 3. Follow the [compass calibration procedure](../config/compass.md#performing-the-calibration)
> 
> After this, you should have a calibrated magnetometer with a custom specified rotation.

Reviewer: ```suggestion
You can use a custom orientation by first selecting `Custom Euler Angle` in the orientation setting for your magnetometer.  
You can then set any rotation using the [CAL_MAGn_ROLL](../advanced_config/parameter_reference.md#CAL_MAG1_ROLL), [CAL_MAGn_PITCH](../advanced_config/parameter_reference.md#CAL_MAG1_PITCH) and [CAL_MAGn_YAW](../advanced_config/parameter_reference.md#CAL_MAG1_YAW) parameters.

The roll/pitch/yaw are defined as documented [here](../config/flight_controller_orientation.md#orientation-definition).

To configure a custom orientation and calibrate the compass:

1. Set the [CAL_MAGn_ROT](../advanced_config/parameter_reference.md#CAL_MAG1_ROT) parameter to `Custom Euler Angle`
2. Set the magnetometer's correct custom roll/pitch/yaw
3. Follow the [compass calibration procedure](../config/compass.md#performing-the-calibration)
```

If you're using the normal [Compass Calibration](../config/compass.md) process (with parameter [CAL_MAG_ROT_AUTO](../advanced_config/parameter_reference.md#CAL_MAG_ROT_AUTO) enabled), the orientation should be detected automatically.
Otherwise you can directly select the appropriate value in [CAL_MAGn_ROT](../advanced_config/parameter_reference.md#CAL_MAG1_ROT) for up to three compasses.

> If your compass can't be oriented in any of the standard orientations, you can use a [custom rotation](#custom_orientation) as documented below.

Reviewer: FYI you'll note that this is different from last time you saw it. I made some changes here, and then liked them so much I applied them immediately inhttps://github.com/PX4/PX4-user_guide/commit/f2428a3aa82910c9bd08a35cb0effcbe802a2f92 - and then fixed the merge conflict.

This is not just the variation against current main.

Author: Oh gotcha!
